rust-runtime/aws-smithy-client/src/retry/standard.rs

```diff
@@ -0,0 +1,597 @@
+                self.shared
+                    .quota_release(self.local.last_quota_usage, &self.config);
+                None
+            }
+            RetryKind::Error(err) => self.handle_retryable_error(err),
+            _ => None,
+        }
+    }
+
+    /// Determine the correct response given `retry_kind`
+    ///
+    /// If a retry is specified, this function returns `(next, backoff_duration)`
+    /// If no retry is specified, this function returns None
+    fn handle_retryable_error(&self, error_kind: &ErrorKind) -> Option<(Self, Duration)> {
+        let quota_used = {
+            if self.local.attempts == self.config.max_attempts {
+                return None;
+            }
+            self.shared.quota_acquire(error_kind, &self.config)?
+        };
+        let backoff = calculate_exponential_backoff(
+            // Generate a random base multiplier to create jitter
+            (self.config.base)(),
+            // Get the backoff time multiplier in seconds (with fractional seconds)
+            self.config.initial_backoff.as_secs_f64(),
+            // `self.local.attempts` tracks number of requests made including the initial request
+            // The initial attempt shouldn't count towards backoff calculations so we subtract it
+            self.local.attempts - 1,
+        );
+        let backoff = Duration::from_secs_f64(backoff).min(self.config.max_backoff);
+        let next = RetryHandler {
+            local: RequestLocalRetryState {
+                attempts: self.local.attempts + 1,
+                last_quota_usage: Some(quota_used),
+            },
+            shared: self.shared.clone(),
+            config: self.config.clone(),
+            sleep_impl: self.sleep_impl.clone(),
+        };
+
+        Some((next, backoff))
+    }
+}
+
+impl<Handler, R, T, E> tower::retry::Policy<Operation<Handler, R>, SdkSuccess<T>, SdkError<E>>
+    for RetryHandler
+where
+    Handler: Clone,
+    R: ClassifyRetry<SdkSuccess<T>, SdkError<E>>,
+{
+    type Future = BoxFuture<Self>;
+
+    fn retry(
+        &self,
+        req: &Operation<Handler, R>,
+        result: Result<&SdkSuccess<T>, &SdkError<E>>,
+    ) -> Option<Self::Future> {
+        let classifier = req.retry_classifier();
+        let retry_kind = classifier.classify_retry(result);
+        self.retry_for_kind(retry_kind)
+    }
+
+    fn clone_request(&self, req: &Operation<Handler, R>) -> Option<Operation<Handler, R>> {
+        req.try_clone()
+    }
+}
+
+fn check_send<T: Send>(t: T) -> T {
+    t
+}
+
+#[cfg(test)]
+mod test {
+    use std::time::Duration;
+
+    use aws_smithy_types::retry::{ErrorKind, RetryKind};
+
+    use crate::retry::NewRequestPolicy;
+
+    use super::{calculate_exponential_backoff, Config, Policy, RetryHandler};
+
+    fn test_config() -> Config {
+        Config::default().with_base(|| 1_f64)
+    }
+
+    #[test]
+    fn retry_handler_send_sync() {
+        fn must_be_send_sync<T: Send + Sync>() {}
```

Comment:
This is cool. I know there is a crate out there called [static-assertions](https://github.com/nvzqz/static-assertions-rs), but what we have here will do.